### PR TITLE
Updated for Postgres crud operations

### PR DESCRIPTION
**Pull request by ChasVanDav**
I installed dependencies and re-organized the file structure so there are a lot of new files to accommodate dependencies and default configuration. 

The main changes can be seen where index.js and book.js are swapped for server.js and pg dump of books table of postgresql database with those crud operations outlined in server.js.

### CODE REVIEW

File: 6-rest-api/frontend/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+// https://vite.dev/config/
+export default defineConfig({
+  plugins: [react()],
```
Comment (daaimah123) [inline 6-rest-api/frontend/vite.config.js:6]: Is your project running okay without a proxy set up?

Comment (ChasVanDav) [reply 6-rest-api/frontend/vite.config.js:6]: Screenshot 2024-10-28 at 2 48 21 PM

Comment (ChasVanDav) [reply 6-rest-api/frontend/vite.config.js:6]: Yes, it is running smoothly. I had played around with vite.config, but it was working as long as I had the backend url clearly referenced in the frontend.


Screenshot 2024-10-28 at 2 49 59 PM

**Review by daaimah123 — APPROVED** [review]
LGTM

**Pull request merged**